Client/Frontend/Settings/ThemeSettings/ThemeMiddleware.swift

```diff
@@ -2,47 +2,84 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at http://mozilla.org/MPL/2.0/
 
-import Foundation
+import Common
 import Redux
 
 protocol ThemeManagerProvider {
     var systemThemeIsOn: Bool { get }
-    func setSystemTheme(isOn: Bool)
-    func setAutomaticBrightness(isOn: Bool)
-    func brightnessChanged(_ value: Float)
 }
 
 class ThemeManagerMiddleware: ThemeManagerProvider {
-    var themeManager: LegacyThemeManager
+    var legacyThemeManager: LegacyThemeManager
+    var themeManager: ThemeManager
 
     var systemThemeIsOn: Bool {
-        themeManager.systemThemeIsOn
+        legacyThemeManager.systemThemeIsOn
     }
 
-    init(themeManager: LegacyThemeManager = LegacyThemeManager.instance) {
-        // TODO: Add support for themeManager
+    init(legacyThemeManager: LegacyThemeManager = LegacyThemeManager.instance,
+         themeManager: ThemeManager = AppContainer.shared.resolve()) {
+        self.legacyThemeManager = legacyThemeManager
         self.themeManager = themeManager
     }
 
     lazy var setSystemTheme: Middleware<AppState> = { state, action in
         switch action {
+        case ThemeSettingsAction.fetchThemeManagerValues:
+            let currentThemeState = self.getCurrentThemeManagerState()
+            DispatchQueue.main.asyncAfter(deadline: .now() + 0.1) {
```

Comment:
Why do we need dispatch after `.now() + 0.1`?

Reply:
This is a hack solution, another point to discuss with Orla and is needed at the moment to ensure the action generated by the middleware is treated after the action generated by the user

Reply:
Removed delay, but we will use the convention of triggering the middleware action using `DispatchQueue.main.async ` which fixes the issue also.